singly_linked_list

```diff
@@ -0,0 +1,317 @@
+//ADT implementation: https://runestone.academy/ns/books/published/pythonds3/BasicDS/TheUnorderedListAbstractDataType.html
+//except for add_item as it copying append
+
+#include <stdio.h>
+#include <stdlib.h>
+
+//define MAX_SIZE; not handled but keep in mind
+
+struct Node
+{
+    int data;
+    struct Node *next;
+};
+
+static struct Node *head;
+
+
+struct Node *List(void) {
+    struct Node *p = (struct Node *) malloc(sizeof(struct Node));
+    p -> data = -1;
+    p -> next = NULL;
+    return p;
+}
+
+int is_empty() {
+    if(head -> data == -1) {
```

Comment:
I suggest to make `-1`a valid list element. In other words an empty list is a list with a head pointer == NULL